### PR TITLE
fix cond x inline_inbuilt

### PR DESCRIPTION
**Pull request by ydwu4**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #134721

TLDR; this PR supports exporting cond x inine_inbuilt nn modules flag by  (internally, the pattern is make_fx(record_module_stack)(torch.compile(f)))

We need to fix two things:

1. _ModuleStackTracer will wrap all the nn modules into _AttrProxy. This _AttrProxy has several hard to inline issues in dynamo like overriding _modules with a property method and overrides the __getattr__,  which mutates captured states when calling `__getattr__`.

Solution to this is that we unwrap the _AttrProxy and get its corresponding nn_module (a 1-1 correspondence). So that dynamo symbolically traces the original nn module instead of tracing _AttrProxy.

2. make_fx will apply a bunch of patches the __getattr__ and __call__ of nn_modules for tracking. This also doesn't work well with dynamo. The immediate error we see is `torch._dynamo.exc.Unsupported: 'inline in skipfiles: WeakKeyDictionary.__contains__ | __contains__ /home/yidi/.conda/envs/pytorch/lib/python3.10/weakref.py`. 

Solution to this is that we remove the patches during dynamo symbolic convert temporally. So that dynamo has a clean environment. make_fx will be tracing the transformed bytecode of dynamo instead.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

### CODE REVIEW

**Comment by ydwu4** [comment]
Replaced by https://github.com/pytorch/pytorch/pull/133731
